### PR TITLE
Make supported versions a configurable field

### LINKED ISSUE #168: Make supported versions a configurable field
Right now OSP supported versions is hardcoded value - it may make more sense to have this as an configurable value in `config.yaml`

### PR DESCRIPTION
**Pull request by ciecierski**
Closes #168 
Signed-off-by: ciecierski <mciecier@redhat.com>

### CODE REVIEW

File: config.yaml.example [hunk 1]
```diff
@@ -34,3 +34,4 @@ stage_logs:
     - '.sh/ir-tripleo-ffu-undercloud-rhel-upgrade.log'
     - '.sh/ir-tripleo-undercloud-rhel-upgrade.log'
     - '.sh/ir-tripleo-undercloud-ffu-upgrade.log'
+supported_versions: ['13', '16.1', '16.2']
```
Comment (nathan-weinberg) [inline config.yaml.example:37]: Put above the stage log stuff

File: docs/README.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ Create a file named `config.yaml` based off `config.yaml.example` with the follo
 - **email_to**: Email address you would like to send your report to. To send the report to multiple emails, seperate them by comma, e.g. recipient1@website1.com,recipient2@website2.org
 - **email_to_test**: Email address to send test reports to (note: this field is only required if you run Jeeves with the `--test-email` flag)
 - **stage_logs**: Optional dict field that instructs Jeeves how to build URLs to a log files for a corresponding failed build stage
+- **supported_versions**: Optional list that instructs Jeeves which osp versions are supported and should
```
Comment (nathan-weinberg) [inline docs/README.md:26]:
```suggestion
- **supported_versions**: Optional list that instructs Jeeves which OSP versions are supported and should
```

File: jeeves/jobs.py [hunk 3]
```diff
@@ -119,8 +119,8 @@ def get_jenkins_job_info(server, job_name, filter_param_name=None, filter_param_
 	return jenkins_api_info
 
 
-def get_jenkins_jobs(server, job_search_fields):
-	''' takes in a Jenkins server object and job_search_fields string
+def get_jenkins_jobs(server, job_search_fields, supported_versions):
+	''' takes in a Jenkins server object, job_search_fields string and supported_versions list
```
Comment (nathan-weinberg) [inline jeeves/jobs.py:123]:
```suggestion
	''' takes in a Jenkins server object, job_search_fields string, and supported_versions list
```

File: config.yaml.example (hunk 1 above)
Comment (nathan-weinberg) [reply config.yaml.example:37]: Furthermore, I feel like having this as a raw Python type input is very limiting and verbose. Why not do a more user-friendly approach?

File: jeeves/report.py [hunk 4]
```diff
@@ -14,8 +14,10 @@
 
 def run_report(config, blockers, preamble_file, template_file, no_email, test_email, server, header):
 
+	# get  supported versions from config or use default one
+	supported_versions = config.get('supported_versions', ['13', '16.1', '16.2'])
 	# fetch all relevant jobs
-	jobs = get_jenkins_jobs(server, config['job_search_fields'])
+	jobs = get_jenkins_jobs(server, config['job_search_fields'], supported_versions)
```
Comment (nathan-weinberg) [inline jeeves/report.py:20]: And what about reminders?

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]

File: jeeves/report.py (hunk 4 above)
Comment (ciecierski) [reply jeeves/report.py:20]: This function is not used in reminder. If a job has owner assigned, it lands in reminder mail. There is no restriction for OSP version in reminder mode.

File: config.yaml.example (hunk 1 above)
Comment (nathan-weinberg) [reply config.yaml.example]: I like it!

File: docs/README.md [hunk 5]
```diff
@@ -22,6 +22,7 @@ Create a file named `config.yaml` based off `config.yaml.example` with the follo
 - **email_from**: The email address of the sender
 - **email_to**: Email address you would like to send your report to. To send the report to multiple emails, seperate them by comma, e.g. recipient1@website1.com,recipient2@website2.org
 - **email_to_test**: Email address to send test reports to (note: this field is only required if you run Jeeves with the `--test-email` flag)
+- **supported_versions**: Optional OSP versions list that instructs Jeeves which OSP versions are supported and should appear in the report
```
Comment (nathan-weinberg) [inline docs/README.md:25]: Maybe add a bit about which versions are supported by default

File: jeeves/report.py [hunk 6]
```diff
@@ -14,8 +14,10 @@
 
 def run_report(config, blockers, preamble_file, template_file, no_email, test_email, server, header):
 
+	# get  supported versions from config or use default one
```
Comment (nathan-weinberg) [inline jeeves/report.py:17]:
```suggestion
	# get supported versions from config or use default one
```

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]
Two nits but other than that LGTM, will test w/ Enterprise jobs next week

**Review by nathan-weinberg — APPROVED** [review]
Works for Enterprise, would like to see some other DFGs test before merging

**Comment by agarcia64** [comment]
Tested on my local machine and things are not working as expected. When I set
supported_versions:
  - 10
  - 13
  - 16.1
  - 16.2
Jeeves failed to include the osp 10 jobs in the report. In addition to this, when I remove 16.1 from the supported versions, the osp 16.1 jobs are still included in the report.

**Review by agarcia64 — CHANGES_REQUESTED** [review]
https://github.com/nathan-weinberg/jeeves/pull/206#issuecomment-827005157

**Review by agarcia64 — APPROVED** [review]

**Pull request merged**